streams/src/main/java/org/apache/kafka/streams/processor/internals/PartitionGroup.java

```diff
@@ -75,6 +76,8 @@ public int compare(final RecordQueue queue1, final RecordQueue queue2) {
         this.partitionQueues = partitionQueues;
 
         totalBuffered = 0;
+        
+        computeTimestamp();
```

Comment:
We can simply the `computeTimestamp()` function here, this is the reasoning:

1) when the task was first initialized, all partition queue's time is `-1`, and as long as any one of the partition is still empty, its queue time would be `-1`, and hence the stream time would be `-1`. Therefore, line 80 here the computed stream time will always be -1 since all record queue are empty.

2) in line 122 below, the stream time would only change if the record queue was always empty before.

So the three callers of `computTimestamp` may be further optimized. I.e. we 0) always set stream time to -1 in constructor, and 2) only need to advance stream time in `nextRecord` if the polled partition represents the current stream time, and 3) only need to advance stream time in `addRawRecords` if it was the first time records are added to a partition queue. We need to add a few more bookkeeping variables for 1) and 2).